src/main/java/org/apache/commons/cli/TypeHandler.java

```diff
@@ -19,190 +19,306 @@ Licensed to the Apache Software Foundation (ASF) under one or more
+    }
+
+    /**
+     * Gets the converter for the the Class. Never null.
+     * @param clazz The Class to get the Converter for.
+     * @return the registered converter if any, {@link Converter#DEFAULT} otherwise.
+     * @since 1.7
+     */
+    public static Converter<?> getConverter(Class<?> clazz) {
+        Converter<?> converter = converterMap.get(clazz);
+        return converter == null ? Converter.DEFAULT : converter;
+    }
+
+    /**
+     * Gets the verifier for the Class. Never null.
+     * @param clazz the Class to get the Verifier for.
+     * @return the registered verifier if any, {@link Verifier#DEFAULT} otherwise.
+     * @since 1.7
+     */
+    public static Verifier getVerifier(Class<?> clazz) {
+        Verifier verifier = verifierMap.get(clazz);
+        return verifier == null ? Verifier.DEFAULT : verifier;
+    }
+
     /**
      * Returns the class whose name is {@code className}.
      *
-     * @param className the class name
-     * @return The class if it is found
-     * @throws ParseException if the class could not be found
+     * @param      className      the class name
+     * @return                    The class if it is found
+     * @throws     ParseException if the class could not be found
+     * @deprecated     use {@link #createValue(String, Class)}
      */
+    @Deprecated // since 1.7
     public static Class<?> createClass(final String className) throws ParseException {
-        try {
-            return Class.forName(className);
-        } catch (final ClassNotFoundException e) {
-            throw new ParseException("Unable to find the class: " + className);
-        }
+        return createValue(className, Class.class);
     }
 
     /**
-     * Returns the date represented by {@code str}.
-     * <p>
-     * This method is not yet implemented and always throws an {@link UnsupportedOperationException}.
+     * Returns the date represented by {@code str}. <p> This method is not yet
+     * implemented and always throws an {@link UnsupportedOperationException}.
      *
-     * @param str the date string
-     * @return The date if {@code str} is a valid date string, otherwise return null.
-     * @throws UnsupportedOperationException always
+     * @param      str the date string
+     * @return         The date if {@code str} is a valid date string, otherwise
+     *                 return null.
+     * @deprecated     use {@link #createValue(String, Class)}
      */
+    @Deprecated // since 1.7
     public static Date createDate(final String str) {
-        throw new UnsupportedOperationException("Not yet implemented");
+        try {
+            return createValue(str, Date.class);
+        } catch (ParseException e) {
+            throw new RuntimeException(e);
+        }
     }
 
     /**
      * Returns the File represented by {@code str}.
      *
-     * @param str the File location
-     * @return The file represented by {@code str}.
+     * @param      str the File location
```

Comment:
Please revert the formatting changes. It makes the PR noisier, larger and takes longer to review.